### DISCUSSION TITLE
Tracing Rust. ERROR: cannot find map files

### DISCUSSION THREAD

**Discussion by ple1n** [namhyung/uftrace#1800] [Q&A]
I built it with `RUSTFLAGS='-C force-frame-pointers=yes -Zinstrument-mcount -Cpasses=ee-instrument<post-inline>' cargo b`

```lof
^Cuftrace: /home/hash/.cache/yay/uftrace/src/uftrace-0.14/cmds/record.c:1437:update_session_maps
 ERROR: cannot find map files: No such file or directory
```

I made uftrace SUID. It seems to be the cause

**Comment by honggyukim** [comment]
Your description looks incomplete so I don't have an idea what your problem is. Could you explain more in details?

**Reply by ple1n** [reply]
I think the cause is exactly that I set it to SUID, by `chmod +s $(which uftrace)`. It works perfectly without SUID, but my app requires SUID to work. 

It ran the app and I could see my app logging, and when I pressed ctrl+c to get results, it said `ERROR: cannot find map files: No such file or directory`. Perhaps uftrace isn't designed to work as SUID ? 

The command I used was `uftrace record -a ./target/debug/nsproxy`. 

I tried both the version from arch repo and compiling from master

**Comment by namhyung** [comment]
Right, `LD_PRELOAD` has limitation with SUID.  See https://man7.org/linux/man-pages/man8/ld.so.8.html.
